cmd/kube-apiserver/app/server.go

```diff
@@ -366,11 +366,21 @@ func (s *APIServer) Run(_ []string) error {
 	s.verifyClusterIPFlags()
 
 	// If advertise-address is not specified, use bind-address. If bind-address
-	// is not usable (unset, 0.0.0.0, or loopback), setDefaults() in
-	// pkg/master/master.go will do the right thing and use the host's default
+	// is not usable (unset, 0.0.0.0, or loopback), we will use the host's default
 	// interface.
 	if s.AdvertiseAddress == nil || s.AdvertiseAddress.IsUnspecified() {
-		s.AdvertiseAddress = s.BindAddress
+		bindAddress := s.BindAddress
+		for bindAddress == nil || bindAddress.IsUnspecified() || bindAddress.IsLoopback() {
+			hostIP, err := util.ChooseHostInterface()
+			if err != nil {
+				glog.Fatalf("Unable to find suitable network address.error='%v' . "+
+					"Will try again in 5 seconds. Set the public address directly to avoid this wait.", err)
+				time.Sleep(5 * time.Second)
```

Comment:
We should change this loop to not be a loop, and this log-lien to not say it will retry - I don't see how that can be true